src/scripts/antithesis.zig

```diff
@@ -0,0 +1,241 @@
+//! Deployment script for our systest (src/testing/systest).
+//!
+//! * Builds the Java client and the associated workload using Maven
+//! * Builds Docker images for the workload, replicas, and config
+//! * Optionally pushes the images to the Antithesis registry
+//!
+//! Currently there's no support for triggering tests with this script.
+
+const std = @import("std");
+const builtin = @import("builtin");
+const log = std.log;
+const assert = std.debug.assert;
+
+const stdx = @import("../stdx.zig");
+const flags = @import("../flags.zig");
+const fatal = flags.fatal;
+const Shell = @import("../shell.zig");
+
+pub const CLIArgs = struct {
+    tag: []const u8,
+    push: bool = false,
+};
+
+const Image = enum { config, workload, replica };
+
+pub fn main(shell: *Shell, gpa: std.mem.Allocator, cli_args: CLIArgs) !void {
+    assert(try shell.exec_status_ok("docker --version", .{}));
+
+    // Docker tag to build and push
+    const tag = std.mem.trim(u8, cli_args.tag, &std.ascii.whitespace);
+    assert(tag.len > 0);
```

Comment:
I'd say rather than trimming, it would be more tiger-style to assert that there aren't whitespace.